addons/stock/models/product.py

```diff
@@ -629,6 +629,11 @@ def _get_rules_from_location(self, location, route_ids=False, seen_rules=False):
 
     def _get_dates_info(self, date, location, route_ids=False):
         rules = self._get_rules_from_location(location, route_ids=route_ids)
+        if self.env.context.get('exclude_transit_location') and any(rule.location_dest_id.usage == 'transit' for rule in rules):
```

Comment:
I think we actually need to fine-tune here
Unfortunately, there might be a path through transit locations that doesn't go in another stock location (weird but possible)
So... Let's come back to one of the ideas we had:
Ensure that none of the source locations will match the condition again (i.e. source loc that is a child of warehouse stock loc)
This is not a lot of change, just an update of the `any`